packages/components/src/internal/components/domainproperties/DomainForm.tsx

```diff
@@ -664,12 +687,46 @@ export class DomainFormImpl extends React.PureComponent<IDomainFormInput, IDomai
         }
     };
 
+    importFieldsFromJson = (file: File): Promise<SimpleResponse> => {
+        const { domain, onChange } = this.props;
+
+        return file.text()
+            .then(text => {
+                const domainType = domain.domainType;
+                const jsFields = JSON.parse(text);
+                if (jsFields.length < 1) return {success: false, msg: 'No fields found.'};
+
+                for (let i=0; i < jsFields.length; i++){
+                    let field = jsFields[i];
+
+                    if (field.defaultValueType && !domain.hasDefaultValueOption(field.defaultValueType)) {
+                        return {success: false, msg: `Error on importing field '${field.name}': Default value '${field.defaultValueType}' is invalid.`};
+                    }
+
+                    if (domainType !== 'list' && field.lockType === DOMAIN_FIELD_PRIMARY_KEY_LOCKED) {
```

Comment:
In my very quick local testing, this domainType was always undefined so this check always failed, even when importing a list field json into a new list.

Reply:
That's kind of weird. I'm in-product on `0.99.0-fb-fieldEditorExportImport-1.1` and list-to-list imports work fine. There was a bug in how I'd changed the domainType to capital 'List' and was still catching for undercase 'list' still in this check, but PKs are now being blocked again in non-PKs as intended. Could you pull and re-check this, and we'll meet if you're still seeing the issue?

Reply:
Update: Implemented change on server-side